runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssOverlay.java

```diff
@@ -128,35 +125,61 @@ private void renderRifts(Graphics2D graphics, DecorativeObject object)
 				graphics.draw(objectClickbox);
 				graphics.setColor(new Color(255, 0, 255, 20));
 				graphics.fill(objectClickbox);
+				BufferedImage overlayImage = getOverlayImage(rift);
+
+				if (config.showOverlayRifts())
+				{
+					OverlayUtil.renderClickboxOverlayImage(graphics, objectClickbox, overlayImage);
+				}
 			}
 		}
 
 		//Draw minimap
-		BufferedImage image = getImage(rift);
-		Point miniMapImage = Perspective.getMiniMapImageLocation(client, object.getLocalLocation(), image);
+
+		if (!config.showMinimapRifts())
+		{
+			return;
+		}
+
+		BufferedImage minimapImage = getMinimapImage(rift);
+		Point miniMapImage = Perspective.getMiniMapImageLocation(client, object.getLocalLocation(), minimapImage);
 
 		if (miniMapImage != null)
 		{
-			graphics.drawImage(image, miniMapImage.getX(), miniMapImage.getY(), null);
+			graphics.drawImage(minimapImage, miniMapImage.getX(), miniMapImage.getY(), null);
 		}
 	}
 
-	public BufferedImage getImage(AbyssRifts rift)
+	public BufferedImage getOverlayImage(AbyssRifts rift)
+	{
+		BufferedImage image = abyssOverlayIcons.get(rift);
+		if (image != null)
+		{
+			return image;
+		}
+
+		image = itemManager.getImage(rift.getItemId());
```

Comment:
The item manager already has its own cache so there is no reason to do this here.